### PR TITLE
add abstraction code sample and documentation to object oriented programming folder

### PR DESCRIPTION
**Pull request by ashan-tharaka**
I created abstraction folder in oop folder and add to it sample abstraction code and documentation.

### CODE REVIEW

File: 1. Object Oriented Programming/2.abstraction/introduction/sample.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public abstract void animalSound();
+}
+
+// Subclass (inherit from Animal)
+class Dog extends Animal {
+    public void animalSound() {
+        // The body of animalSound() is provided here
+        System.out.println("The Dog says: bow bow");
+    }
+}
```
Comment (buddhikac96) [inline 1. Object Oriented Programming/2.abstraction/introduction/sample.java]: Add an empty line to the end of the file

File: 1. Object Oriented Programming/2.abstraction/introduction/doc.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Abstract class: is a restricted class that cannot be used to create objects (to access it, it must be inherited from another class).
+
+Abstract method: can only be used in an abstract class, and it does not have a body. The body is provided by the subclass (inherited from).
+
+#####● Abstraction means only essentials details display to the user.
+#####● Consider a TV. In the user manual they only mention how to change channels, not what happen in the TV circuits, when user change a chanel.
+#####● Abstraction reduce the complexity of the code.
+#####● Avoid code duplication and increase the re-usability.
+#####● Helps to increase security of an application or program as only important details are provided to the user.
```
Comment (buddhikac96) [inline 1. Object Oriented Programming/2.abstraction/introduction/doc.md]: Please use your code also to explain the concept

File: 1. Object Oriented Programming/2.abstraction/introduction/sample.java (hunk 1 above)
Comment (ashan-tharaka) [reply 1. Object Oriented Programming/2.abstraction/introduction/sample.java]: ok @buddhikac96

File: 1. Object Oriented Programming/2.abstraction/introduction/doc.md (hunk 2 above)
Comment (ashan-tharaka) [reply 1. Object Oriented Programming/2.abstraction/introduction/doc.md]: ok @buddhikac96

**Review by ashan-tharaka — COMMENTED** [review]
add changes

**Pull request merged**